component/loki/source/podlogs/internal/apis/monitoring/v1alpha2/types.go

```diff
@@ -0,0 +1,41 @@
+package v1alpha2
+
+import (
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path="podlogs"
+// +kubebuilder:resource:path="podlogs"
+// +kubebuilder:resource:categories="grafana-agent"
+
+// PodLogs defines how to collect logs for a Pod.
+type PodLogs struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PodLogsSpec `json:"spec,omitempty"`
+}
+
+// PodLogsSpec defines how to collect logs for a Pod.
+type PodLogsSpec struct {
+	// Selector to select Pod objects. Required.
+	Selector metav1.LabelSelector `json:"selector"`
```

Comment:
The selector docstring says the Selector is required in a PodLogs spec, but we have this block as optional. Or is it a different thing? cc @rfratto

Reply:
Yeah, nice catch, we should add a kubebuilder annotation to say this is required for anyone writing a PodLogs resource (we do this in some of the operator types if you need a reference on how to do that). I don't think the selectors on the Flow configs need to be required though.